sklearn/ensemble/tests/test_iforest.py

```diff
@@ -129,6 +130,12 @@ def test_iforest_error():
     with pytest.raises(NotImplementedError, match=msg):
         IsolationForest(behaviour='old').fit(X)
 
+    # test feature_weight length matches n_features:
```

Comment:
Instead of a comment, please use a variant of assert_raises that allows you to check the error message, such as `pytest.raises(ValueError, match=...)`